tests/unit/design_document_tests.py

```diff
@@ -781,6 +799,27 @@ def test_get_search_info(self):
                               'pending_seq': 101, 'committed_seq': 0},
              })
 
+    @unittest.skipUnless(
+        os.environ.get('RUN_CLOUDANT_TESTS') is not None,
+        'Skipping Cloudant _search_info raises HTTPError test'
+    )
+    def test_get_search_info_raises_httperror(self):
+        """
+        Test get_search_info raises an HTTPError.
+        """
+        # Mock HTTPError when running against Cloudant
+        search_index = 'search001'
+        resp = requests.Response()
+        resp.status_code = 400
+        self.client.r_session.get = mock.Mock(return_value=resp)
+        with self.assertRaises(requests.HTTPError) as cm:
+            ddoc = DesignDocument(self.db, '_design/ddoc001')
```

Comment:
I would move this line before the `with self.assertRaises...` that way it ensures that the HTTPError is being raised on the call to `.search_info(...)`.